lib/formats/bs_format_ripejson.c

```diff
@@ -220,6 +226,49 @@ static int process_common_fields(bgpstream_format_t *format,
   return 0;
 }
 
+// readline
+static int readline(bgpstream_format_t *format)
+{
+  int total_read = 0;
+  int current_read = 0;
+  char *buffer_ptr = STATE->json_string_buffer;
+
+  while(1){
+    // read line, including linebreak
+    current_read = bgpstream_transport_readline(
+      format->transport, buffer_ptr, BUFSIZE - total_read, 0);
+    total_read+=current_read;
+
+    if(current_read == 0){
+      // reach EOF, no bytes read, return 0
+      return total_read;
+    }
+
+    if(current_read < BUFSIZE - total_read -1 ||
+      buffer_ptr[current_read-1] == '\n'){
+      // if read less than buffer, or last byte is an newline, readline is finished
+
+      // replace linebreak with null
+      if(buffer_ptr[current_read-1] == '\n'){
+        buffer_ptr[current_read-1] = '\0';
+        total_read--;  // decrease the total reads count by one for the newline
+      }
+      break;
+    }
+
+    // if reaches here: no newline received yet, need to read more
+    // reallocate memory
+    BUFSIZE += JSON_BUFFER_BLOCK_SIZE;
+    STATE->json_string_buffer = (char*) realloc (STATE->json_string_buffer, BUFSIZE * sizeof(char));
+    // point temporary buffer to the last read location
+    buffer_ptr = &STATE->json_string_buffer[total_read - 1];
```

Comment:
- [ ] `total_read -1` should be `total_read`
- [ ] examine why current tests passed for `total_read-1`

Reply:
`bgpstream_transport_readline` does not always null-terminate.
`./tools/bgpreader -d singlefile -o upd-type=ripejson -o upd-file=test/ris-live-stream.json`

```
current_read = bgpstream_transport_readline(
      format->transport, buffer_ptr, BUFSIZE - total_read, 0);
    total_read+=current_read;
    assert(STATE->json_string_buffer[total_read] == '\0');
```
```
Assertion failed: (STATE->json_string_buffer[total_read] == '\0'), function readline
```